### PR TITLE
Add api versioning

### LINKED ISSUE #219: Api Versioning
We should start versioning the api. 

As a starting point, all routes should be amended with the prefix `/api/v1/`.

Plenty of information [here](https://github.com/microsoft/aspnet-api-versioning/wiki)

### PR DESCRIPTION
**Pull request by dandanlen**
This adds api versioning.
For now there is just a version 1 - we can create version 2 once version 1 is frozen. 

Note the new base path is `/api/v1/`.

Fixes #219

### CODE REVIEW

File: tests/explorer.api.tests/ApiTests.cs [hunk 1]
```diff
@@ -166,7 +171,7 @@ public async Task SuccessWithResult()
         public async Task FailWithBadEndPoint(string endpoint)
         {
             await TestApi(HttpMethod.Post, endpoint, ValidData, test: (response, content) =>
-                Assert.True(response.StatusCode == HttpStatusCode.NotFound, content));
+                Assert.False(response.IsSuccessStatusCode, content));
```
Comment (AndreiBozantan) [inline tests/explorer.api.tests/ApiTests.cs:174]: The original assert contained additional information about the status code.

Comment (dandanlen) [reply tests/explorer.api.tests/ApiTests.cs:174]: Yeah this was a bit lazy. I did it because the api extension returns a BadResponse (400) or NotAllowed (405) instead of NotFound (404). So depending on whether the test case hits the api version checker or our api endpoint we see different response codes. This was just a convenient catch-all. 

I suppose I shouldn't be testing the api versioning here (that's an external package after all and I assume it's tested by Microsoft 😉). I'll add the `/api/v1` prefix to the test urls.

**Comment by dandanlen** [comment]
There's a long discussion on slack about this. 

TLDR. We can either maintain a separate namespace like `Metrics.V1.MyMetric` and dispatch to the correct one at runtime, or we can maintain separate branches in the repository. 

However we don't need to decide this until we actually freeze api v1.

**Comment by AndreiBozantan** [comment]
Could we also keep the URL's without versions, e.g. `/api/explore`, which will server the latest version?

**Comment by dandanlen** [comment]
Yes that's possible, however apparently not recommended - the client should 'know' what api version they want to target and do this explicitly.

File: tests/explorer.api.tests/ApiTests.cs (hunk 1 above)
Comment (dandanlen) [reply tests/explorer.api.tests/ApiTests.cs]: I've made the tests more specific: 959ea03

File: tests/explorer.api.tests/ApiTests.cs [hunk 2]
```diff
@@ -242,6 +242,16 @@ public async Task FailWithInvalidApiKey()
                 });
         }
 
+        [Fact]
+        public async Task FailWithInvalidExplorationId()
+        {
+            await TestApi(
+                HttpMethod.Get,
+                $"{resultEndpoint}/c29191e7-408e-4395-bb81-660c47181661",
```
Comment (AndreiBozantan) [inline tests/explorer.api.tests/ApiTests.cs:260]: I think that it would be helpful for the semantics to use GUID like "11111111-1111-1111-1111-111111111111".

**Comment by dandanlen** [comment]
@andrei I just force-pushed this with some changes to the README and an extra test case - can you please just check to make sure everything looks ok?

File: tests/explorer.api.tests/ApiTests.cs (hunk 2 above)
Comment (dandanlen) [reply tests/explorer.api.tests/ApiTests.cs]: done 
4da4db3
